### PR TITLE
Invert TabbableContainer into TabbableContentContainer

### PR DESCRIPTION
**Pull request by nyquillerium**
Resolves https://github.com/ppy/osu-framework/issues/2550

The search logic remains unchanged (except to adapt to its new location in the hierarchy), but I have added comments to better describe what it is doing.

This allows any component that should accept tab focus to do so by implementing `ITabbableContainer` instead of inheriting `TabbableContainer`. 

Also adds tests for tabbing.

This requires an osu-side PR to change all usages of `TextBox`

### CODE REVIEW

**Comment by nyquillerium** [comment]
I noticed while making an osu-side PR for this that quite a few textboxes depend on the fact that its a `Container` to apply effects. Not sure if it would be fine to just leave `TextBoxes` as `Container`s for now.

**Comment by peppy** [comment]
Probably needs a test for nested `TabbableContentContainer`s to enforce how they behave.

File: osu.Framework/Graphics/Containers/TabbableContentContainer.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    /// </summary>
+    internal interface ITabTarget
+    {
+        /// <summary>
+        /// Whether this element can be tabbed to.
+        /// </summary>
+        bool CanBeTabbedTo { get; }
+    }
+
+    public class TabbableContentContainer<T> : Container<T>
```
Comment (swoolcock) [inline osu.Framework/Graphics/Containers/TabbableContentContainer.cs]: I really don't like the naming of this class to be honest.  Every time I see it I think it's a tab container, like you'd have in a preferences pane or for document switching.  Something like `FocusSwitchingContainer` or `FocusChainContainer` makes more sense to me.

I can think of at least two situations where the term "tabbing" doesn't make sense:

1) Tapping "next" on a software keyboard
2) Directional focus changing, such as swiping/tapping on a tvOS/Android TV remote, D-pad on a game controller or joystick

File: osu.Framework/Graphics/Containers/TabbableContentContainer.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+        protected override void LoadComplete()
+        {
+            inputManager = GetContainingInputManager();
+            base.LoadComplete();
+        }
+
+        protected override bool OnKeyDown(KeyDownEvent e)
+        {
+            if (e.Key != Key.Tab)
```
Comment (swoolcock) [inline osu.Framework/Graphics/Containers/TabbableContentContainer.cs]: I don't think we should be hardcoding `Key.Tab`, it seems like a candidate for a `PlatformAction` (though I can't think of a platform off-hand that **doesn't** use tab for focus change).

**Comment by smoogipoo** [comment]
Closing due to inactivity. Probably still something we'll want to consider in the future, but is already significantly out of date.

**Pull request closed without merging**